components/service/glean/src/test/java/mozilla/components/service/glean/storages/PingStorageEngineTest.kt

```diff
@@ -0,0 +1,176 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.storages
+
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.Dispatchers
+import mozilla.components.service.glean.FakeDispatchersInTest
+import mozilla.components.service.glean.Glean
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.resetGlean
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertFalse
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import java.io.File
+import java.io.FileReader
+import java.io.BufferedReader
+import java.io.FileOutputStream
+import java.util.UUID
+
+@RunWith(RobolectricTestRunner::class)
+class PingStorageEngineTest {
+    // Filenames and paths to test with
+    private lateinit var fileName1: UUID
+    private lateinit var path1: String
+    private lateinit var fileName2: UUID
+    private lateinit var path2: String
+    private lateinit var fileName3: UUID
+    private lateinit var path3: String
+
+    private lateinit var fileNames: Array<String>
+    private lateinit var pathNames: Array<String>
+
+    private lateinit var pingStorageEngine: PingStorageEngine
+
+    @get:Rule
+    val fakeDispatchers = FakeDispatchersInTest()
+
+    @Before
+    fun setup() {
+        resetGlean()
+        pingStorageEngine = Glean.pingScheduler.pingStorageEngine
+
+        // Clear out pings and assert that there are none in the directory before we start
+        pingStorageEngine.listPingFiles()?.forEach { file ->
+            file.delete()
+        }
+        assertNull("Pending pings directory must be empty before test start",
+            pingStorageEngine.listPingFiles())
+
+        fileName1 = UUID.randomUUID()
+        fileName2 = UUID.randomUUID()
+        fileName3 = UUID.randomUUID()
+
+        path1 = "/test/$fileName1"
+        path2 = "/test/$fileName2"
+        path3 = "/test/$fileName3"
+
+        fileNames = arrayOf(fileName1.toString(), fileName2.toString(), fileName3.toString())
+        pathNames = arrayOf(path1, path2, path3)
+    }
+
+    @Test
+    fun `storage engine correctly stores pings to file`() {
+        runBlocking(Dispatchers.IO) {
+            pingStorageEngine.store(fileName1, path1, "dummy data")
+        }
+        Thread.sleep(10)
```

Comment:
Let's not have `Thread.sleep` stuff, as it will make tests intermittent if CI is under pressure. Let's investigate why this isn't working. Here and in the other tests below.

Reply:
Found a better way in a trick I learned from PR #2120